### DISCUSSION TITLE
Translate achievements title and descriptions

### DISCUSSION THREAD

**Discussion by krupupakku** [RetroAchievements/RAWeb#1110] [Ideas]
Hey there,

Would be wonderful have the possibility to translate title and description for a given trophy/achievement in many languages
This could be done by the community and fallback to english in case of missing translation.

some ideas:
- Translation could be done manually to preserve quality (no auto-translation, it's better fallback to eng),
- Translation should be approved by one or 2 native speaker (or maybe an admin),
- The goal is match your language and show achievements in your favourite language.
- frontends could fetch it adding parameter as ?lang=it

_I searched on RA web, discord, docs, no info about this topic (if it was already discussed)_

What do you think? 
Thank you!!

Example from Tombi! - Tomba! game:

- Eng title: Animal Dash
- Ita title: Scatto Animale

- Eng desc: Acquire the Animal Dash ability from Charles
- Ita desc: Ottieni l'abilità Scatto Animale da Charles

<img width="776" alt="AchievTombi" src="https://user-images.githubusercontent.com/29118355/187456015-a85ae4b3-e407-4340-ad53-1f78c8714a3d.png">

**Comment by Jamiras** [comment]
These values are entered by the achievement creators, most of whom only speak one or two languages (and it's obvious that for some of them English isn't their first language). It would be unreasonable to expect anyone to voluntarily provide translations for the ~250,000 achievements already in the system, and that number is constantly growing. In Q2 of this year, there were 23000 achievements added - that's around 250 per day! 

> Translation could be done manually to preserve quality (no auto-translation, it's better fallback to eng),

There are already many achievements with spelling or grammatical errors, or descriptions that are unclear as to what the requirements for the achievement actually are. What happens when the English text is modified? Do the translations just get cleared out?

> Translation should be approved by one or 2 native speaker (or maybe an admin),

How would the approval process work? What if another user disagreed with your translation? How would conflicts be resolved?

I know we have German, French, Brazilian, and Russian speakers. You've suggested Italian. Some people prefer to play the Japanese games in Japanese (Even if they're not native Japanese speakers, they may want the popups to be in Japanese too). There's nothing saying we couldn't support Greek or Hebrew. Or even Latin. Thankfully, Unicode doesn't support Klingon yet.

What would the UI look like for entering these values? Would it be a giant grid with every possible language? When there is a conflict, how would alternative suggestions be provided/resolved?

> The goal is match your language and show achievements in your favourite language.

I imagine that if we did do this, the 200 or so most popular games would get translations for most languages. Another 1000 games or so would get support for one or two additional languages, and the other 4500 would remain English only.

> I searched on RA web, discord, docs, no info about this topic (if it was already discussed)

#1006 is tracking language support for the website itself, but there's a fixed number of strings there.

> What do you think?

Leaderboard titles/descriptions would clearly fall under the same ruleset as achievements, but what about Rich Presence? Those are calculated client side and shared through the server. If you uploaded an Italian Rich Presence, it would appear as Italian for all users looking at the Active Players widget on the main page or Recent Players widget on the game page. 

I'm not suggesting that we don't do this. I'm just pointing out that there's a lot of stuff to consider before attempting it.

**Reply by krupupakku** [reply]
Hey good points! Thanks for the answer.

> It would be unreasonable to expect anyone to voluntarily provide translations for the ~250,000 achievements already in the system.

yes, you're right. I assume English would remain the primary language and the only one "mandatory", I would consider any extra language as a plus, an enhancement. The author would only implement english stuff and other people could add stuff in the future.

> There are already many achievements with spelling or grammatical errors, or descriptions that are unclear as to what the requirements for the achievement actually are. What happens when the English text is modified? Do the translations just get cleared out?

As you correctly said, English is not the birth language of lots of people (I'm part of this group), so we tend to do errors. I suggest to flag (don't know how atm) as "outdated - translation could be wrong or outdated" a translation when the primary English string is updated.  

> There's nothing saying we couldn't support Greek or Hebrew. Or even Latin. 

eheh I know this could be an infinite and tough task, but maybe we can start with easy languages to implement and common, such as spanish as they use the same writing system.

> and the other 4500 would remain English only.

As at the moment we have only english I don't see the problem. I get translations for translated games and english for the rest. :D 

> What would the UI look like for entering these values? Would it be a giant grid with every possible language? 

In my mind I imagined to show only selected languages and the english one. with a "+ new translation" or something like:
eng string -> add new translation -> it -> it string -> show side by side eng-ita, if I want to provide spanish localization I would change filters and do the same.
I don't have a final solution right now. 

> https://github.com/RetroAchievements/RAWeb/discussions/1006 is tracking language support for the website itself, but there's a fixed number of strings there.

Thanks about this, maybe I can contribute.

-.-.-.-.-.-.-.-.-.- 

I don't have any ideas/solutions about: 
- Rich Presence
- how/who should resolve a conflict?

I hope we can achieve this someday ;)
